device-scripts/diag.sh

```diff
@@ -0,0 +1,186 @@
+            if [ ! -f /var/sota/${FILE} ]
+            then
+                echo "### missing /var/sota/$FILE ###"
+                bad=1
+            fi
+        done
+        if [ $bad -eq 0 ]
+        then
+            echo -e "\n*** checking certs ***"
+           openssl s_client -cert /var/sota/client.pem \
+              -CAfile $ROOT_CERT_DIR \
+              -key /var/sota/pkey.pem \
+              -connect $SERV < /dev/null || echo "### Failed ###"
+        else
+            echo -e "\n*** no cert checking connectivity ***"
+            openssl s_client -connect $SERV < /dev/null || echo "### Failed ###"
+        fi
+    else
+        CLIENT_ID=$(grep "^tls_clientcert_id" $CFG | sed 's/^.*= "//;s/"$//')
+        PKEY_ID=$(grep "^tls_pkey_id" $CFG | sed 's/^.*= "//;s/"$//')
+        if [ "$ROOT_CERT_DIR" = "/var/sota/root.crt" ]
+        then
+            echo -e "\n*** checking HSM certs ***"
+        else
+            echo -e "\n*** checking HSM+EL2GO certs ***"
+        fi
+        cat > /var/tmp/engine.cnf <<EOF
+openssl_conf = oc
+[oc]
+engines = eng
+[eng]
+pkcs11 = p11
+[p11]
+engine_id = pkcs11
+dynamic_path = /usr/lib/engines-3/pkcs11.so
+MODULE_PATH = $MODULE
+init = 0
+PIN = $PIN
+EOF
+        pkcs11-tool --module $MODULE --login \
+            --pin $PIN --token-label aktualizr --id $CLIENT_ID \
+            --read-object --type cert | \
+            openssl x509 -inform DER -outform PEM > /var/tmp/client.pem
+        OPENSSL_CONF=/var/tmp/engine.cnf openssl s_client \
+            -engine pkcs11 \
+            -keyform engine \
+            -key "pkcs11:token=aktualizr;id=%$PKEY_ID;type=private;pin-value=$PIN" \
+            -cert /var/tmp/client.pem -CAfile $ROOT_CERT_DIR \
+            -connect $SERV < /dev/null
+        rm -f /var/tmp/client.pem /var/tmp/engine.cnf
+    fi
+else
+    echo "*** not registered import version ***"
+    cat /var/sota/import/installed_versions
+    echo
+fi
+echo
+
+echo "*** Aktualizr-lite status ***"
+systemctl status aktualizr-lite | cat
+echo
+
+echo "*** Docker images ***"
+docker images
+echo
+
+echo "*** Docker containers ***"
+docker ps -sa
+echo
+
+echo "*** Docker usage ***"
+du -d 1 /var/lib/docker/
+echo
+
+echo "*** Number of zero length files in /var/lib/docker ***"
+find /var/lib/docker -size 0 | wc -l
+echo
+
+echo "*** sha256sum of /var/sota/reset-apps ***"
+sha256sum $(find /var/sota/reset-apps -type f) < /dev/null
+echo
+
+echo "*** Domain access check ***"
+res=0
+for domain in hub.foundries.io ota-lite.foundries.io ostree.foundries.io
+do
+    nslookup "$domain"
+    res=$(($res + $?))
+done
+echo
+
+if [ $res -eq 0 ]
+then
+    echo "*** domain latency ***"
+    for target in hub.foundries.io ota-lite.foundries.io ostree.foundries.io
+    do
+        echo "$target latency $(ping -c 5 "$target" | sed -nE 's|rtt min/avg/max/mdev = [0-9]+.[0-9]+/||;s|/.*$| ms|p')"
+    done
+else
+    echo "*** domain access error no latency ***"
+fi
+echo
+
+echo "*** speed test ***"
```

Comment:
Curious why doing a speed test here? What kind of information are you after?

This will cause the client to reach a service we don't control, which will give different results based on where the device is located and will also cause the device to consume network data.

Reply:
Some problems we have encountered was due to limited bandwidth that is why we added this test.

Reply:
Would it make more sense to have a separate speed test script?

Reply:
I believe that might be better yeah.

Reply:
Or just make an argument and make it optional.